src/settings/DesktopUpdateHelpLabel.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+import m from "mithril"
+import {Icon} from "../gui/base/Icon"
+import {noOp} from "../api/common/utils/Utils"
+import {lang} from "../misc/LanguageViewModel"
+import {BootIcons} from "../gui/base/icons/BootIcons"
+import {nativeApp} from '../native/NativeWrapper.js'
+import {Request} from "../api/common/WorkerProtocol.js"
+
+export type UpdateHelpLabelAttrs = {
+	updateAvailable: Stream<boolean>;
```

Comment:
We never subscribe to this stream inside, I think we should just pass boolean as an attr

Reply:
it's changed from within getOnClick and is then used in the SettingsViewer for the next redraw 
https://github.com/tutao/tutanota/blob/696cff30692ce12b4f58c2a879639927fb73155b/src/settings/DesktopUpdateHelpLabel.js#L64-L76

the attrs type could be Stream<boolean>, but I don't see a way to return the state of the update check to the settings viewer as easily with just a bool.

Reply:
Ah, so it's for propagating back? my bad, I missed it
I think callback is more clear (and I would change more places where we use streams for callbacks in our code) but I don't have a strong opinion about this place.